pallets/connectors/src/message.rs

```diff
@@ -35,34 +35,93 @@ where
 	Rate: Encode + Decode,
 {
 	Invalid,
+	AddCurrency {
+		currency: u128,
+		evm_address: [u8; 20],
+	},
 	AddPool {
 		pool_id: PoolId,
 	},
+	AllowPoolCurrency {
+		currency: u128,
+		pool_id: PoolId,
+	},
 	AddTranche {
 		pool_id: PoolId,
 		tranche_id: TrancheId,
 		token_name: [u8; TOKEN_NAME_SIZE],
 		token_symbol: [u8; TOKEN_SYMBOL_SIZE],
+		decimals: u8,
 		price: Rate,
 	},
-	UpdateTokenPrice {
+	UpdateTrancheTokenPrice {
 		pool_id: PoolId,
 		tranche_id: TrancheId,
 		price: Rate,
 	},
 	UpdateMember {
 		pool_id: PoolId,
 		tranche_id: TrancheId,
-		address: Address,
+		member: Address,
 		valid_until: Moment,
 	},
+	// Bidirectional: Domain must not accept every incoming token.
+	// Sender cannot ensure whether the receiver rejects.
```

Comment:
I feel like this comment belongs at the header of the `Message` type definition to introduce concepts that will be mentioned below; then for each message we can document each message regarding its directionality and other properties. wdyt?

Reply:
I hope my attempt in [4ecddf9](https://github.com/centrifuge/centrifuge-chain/pull/1292/commits/4ecddf9d7515949ea43e8ad65375bfd1174d4ec4) aligns with your vision.